Add documentation for xk6-disruptor (#891)
grafana/k6-docs: src/data/markdown/docs/40 xk6-disruptor/02 API/03 ServiceDisruptor.md

---
title: 'ServiceDisruptor'
excerpt: 'xk6-disruptor: ServiceDisruptor class'
---

> The `ServiceDisruptor` class allows the injection of different types of faults in the pods that back a Kubernetes service. 

Reviewer: ```suggestion
The `ServiceDisruptor` class can inject different types of faults into the pods that back a Kubernetes service. 
```

To construct a `ServiceDisruptor`, use the [ServiceDisruptor() constructor](/javascript-api/xk6-disruptor/api/servicedisruptor/constructor).
 

## Methods

| Method | Description |
| ------ | ----------- |
| [ServiceDisruptor.injectHTTPFaults()](/javascript-api/xk6-disruptor/api/servicedisruptor/injecthttpfaults) | Inject [HTTTP faults](/javascript-api/xk6-disruptor/api/faults/http) in the target Pods|
 

## Example

> The following example shows how to create a disruptor for the `nginx` service and inject a delay of 100ms and a 10% of requests returning a http response code 500. 

Reviewer: ```suggestion
The following example:
- Creates a disruptor for the `nginx` service
- Injects a delay of 100ms and a 10 percent of requests that return an http response code `500`. 
```
